scripts/upload_s3.py

```diff
@@ -0,0 +1,104 @@
+        data = TorchBenchData(
+            torch_date = pytorch_date,
+            relpath = relpath)
+        return (pytorch_date, data)
+
+def gen_commit_history(result_dir: str):
+    dirs = map(lambda x: os.path.join(result_dir, x), sorted(os.listdir(result_dir)))
+    dirs = filter(lambda x: os.path.isdir(x), dirs)
+    results = dict()
+    for d in dirs:
+        json_files = map(lambda x: os.path.join(d, x), sorted(os.listdir(d)))
+        # Don't analyze empty files
+        filtered_json_files = filter(lambda x: os.stat(x).st_size, json_files)
+        for json_file in filtered_json_files:
+            relpath = os.path.relpath(json_file, result_dir)
+            (torch_date, tb_data) = analyze_json_result(json_file, relpath)
+            if torch_date:
+                # Only store the latest data
+                results[torch_date] = tb_data
+    out = []
+    for key in sorted(results.keys()):
+        test = dict()
+        test["id"] = key
+        test["result"] = results[key]
+        out.append(test)
+    return out
+
+def upload_s3_file(key, body_file):
+    with open(body_file, "r") as body:
+        print(f"Uploading file {body_file} to S3 key: {key}")
+        obj = get_S3_object_from_bucket(S3_BUCKET, key)
+        obj.put(Body=body.read())
+
+def should_upload(json_path):
+    # Upload non-empty json files
+    return (json_path.endswith(".json") and os.path.exists(json_path) and os.stat(json_path).st_size)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("--torchbench-result-dir", required=True,
+                    help="Specify the benchmark result directory")
+    parser.add_argument("--gen-index", action="store_true",
+                    help="Regenerate the benchmark index json")
+    parser.add_argument("--upload-s3", action="store_true",
+                    help="Upload the specified directory to Amazon S3")
+    args = parser.parse_args()
+    parent_dir = Path(args.torchbench_result_dir).parent.absolute()
+    index_path = os.path.join(parent_dir, "index.json")
+    # Generate the index.json file to the parent directory
+    if args.gen_index:
+        # Get parent directory 
+        index_json = gen_commit_history(parent_dir)
+        with open(index_path, "w") as out_file:
+            out_file.write(json.dumps(index_json, indent=4, sort_keys=True, cls=TBEncoder))
+    if args.upload_s3:
+        # Upload index file
+        index_key = "torchbench_v0_nightly/index.json"
+        upload_s3_file(index_key, index_path)
+        # Upload the result directory
+        basedir = os.path.basename(args.torchbench_result_dir)
+        result_dir = f"torchbench_v0_nightly/{basedir}/"
+        for result_file in os.listdir(args.torchbench_result_dir):
+            result_key = f"{result_dir}/{result_file}"
+            result_path = os.path.join(args.torchbench_result_dir, result_file)
+            if should_upload(result_path):
```

Comment:
suggest naming 'is_nonempty_json' instead of 'should_upload', as it is more clear what it does

Reply:
Good point!